lcm-cmake/cpack.cmake

```diff
@@ -0,0 +1,68 @@
+# Package release version
+set(PACKAGE_RELEASE_VERSION 1)
+
+# Default CPack generators
+set(CPACK_GENERATOR TGZ STGZ)
+
+# Detect OS type, OS variant and target architecture
+if(UNIX)
+    set(OS_TYPE Linux)
+
+    # Determine distribution
+    execute_process(COMMAND lsb_release -si
+        OUTPUT_VARIABLE LINUX_DISTRO
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # Determine architecture
+    execute_process(COMMAND uname -m
+        OUTPUT_VARIABLE MACHINE_ARCH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # Add CPack generator according to detected distribution
+    if(LINUX_DISTRO STREQUAL "Debian" OR LINUX_DISTRO STREQUAL "Ubuntu" OR LINUX_DISTRO STREQUAL "LinuxMint")
+        list(APPEND CPACK_GENERATOR DEB)
+    endif()
+
+    # Set OS type and ARCH suffix
+    set(OS_TYPE_ARCH_SUFFIX ${OS_TYPE}-${MACHINE_ARCH})
+else()
+    if(WIN32)
+        set(OS_TYPE Win)
+    elseif(APPLE)
```

Comment:
(Hmm, apparently the "m" is not supposed to be capitalized, either... for that matter, you might consider just making everything lower-case: `linux`, `macos`, `win`...)